Split inverse problem documentation

TorkelE opened the pull request:
This creates a new Documentation tab, "Inverse Problems". Right now, the "Parameter Estimation" tutorial is moved from "Catalyst Application" to "Inverse Problems".

If this passes and people are happy, I am good to have this merged, and then I will start adding the documentation outlined in https://github.com/SciML/Catalyst.jl/issues/672 in separate PRs.

isaacsas commented on docs/pages.jl:
```diff
@@ -10,8 +10,8 @@ pages = Any["Home" => "index.md",
-                                           "catalyst_applications/bifurcation_diagrams.md",
-                                           "catalyst_applications/parameter_estimation.md"],
+                                           "catalyst_applications/bifurcation_diagrams.md"],
+            "Inverse Problems" => Any["inverse_problems/parameter_estimation.md"],
             "Example Networks" => Any["example_networks/basic_CRN_examples.md",
                                       "example_networks/hodgkin_huxley_equation.md",
                                       "example_networks/smoluchowski_coagulation_equation.md"],
```
```suggestion
            "Example Networks" => Any["example_networks/basic_CRN_examples.md",
                                                             "example_networks/hodgkin_huxley_equation.md",
                                                             "example_networks/smoluchowski_coagulation_equation.md"],
             "Inverse Problems" => Any["inverse_problems/parameter_estimation.md"],
```
I think it makes more sense to show examples of building models before showing an analysis that uses models.

isaacsas replied:
Otherwise looks good.

TorkelE replied:
I don't think any of the models under "Inverse Problems" would depend on those in "Example Networks". My thought was that the "Inverse Problem" section is basically equivalent to "Catalyst Application" section, i.e. showing how to do certain analysis (but cut off into its own section to avoid creating a really long one).

isaacsas commented:
Maybe we should swap the Catalyst applications and example sections too. The latter is Catalyst-focused, showing how to build more advanced networks just within Catalyst, while the latter is focused on how Catalyst interfaces with other packages. So the new ordering would be

1. Examples
2. Applications
3. Parameter Estimation

I'd still prefer if we keep the first parameter estimation tutorial focused on SciML tooling, in part because such approaches can be used for the other model types we support too, and are the most flexible (but also do require more to be done by the user). Then in the special case of ODE models we can have a tutorial that introduces PETab and explains how it provides a simple interface and more automated way to fit such models.

TorkelE commented:
Ok, let's put the SciML optimization before the petab one. I'm still not convinced by the ordering (as I was envisioning tutorials in Application and Parameter estimation making references to Examples, and also generally consider Example the part people might read last/the part not actually needed to learn how do something). However, so we can move forward with stuff we can merge in that order for a starter.

isaacsas commented:
Sounds good.

TorkelE commented:
I have
- Moved parameter fitting to a new inverse problem section at the end.
- moved the network examples section to a subsection of Catalyst Functionality (called "Model creation examples")

TorkelE commented:
If this sounds good to you @isaacsas I will merge (once tests passes)

TorkelE commented:
Closed by https://github.com/SciML/Catalyst.jl/pull/701
